src/main/scala/viper/gobra/frontend/info/implementation/property/Assignability.scala

```diff
@@ -64,6 +64,7 @@ trait Assignability extends BaseProperty { this: TypeInfoImpl =>
 
         // for ghost types
       case (BooleanT, AssertionT) => true
+      case (PredT(Vector()), AssertionT) => true
```

Comment:
That case is wrong. Only a predicate instance is an assertion.

Reply:
I know understand why you did that. You changed that predicates have a predicate type, instead of a function type that returns an assertion, but this is wrong. It should stay the old way.

Reply:
To give a simple example, the following is not ok, but might work currently:

```
requires acc(x) && p
func foo(x *int, p pred())
```

Reply:
Yes, you are right! I will revert all related changes